lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -563,7 +562,100 @@ int amf_smf_send(
   }
   return rc;
 }
+/***************************************************************************
+**                                                                        **
+** Name:    amf_smf_validate_ie_params()                                  **
+**                                                                        **
+** Description: function to handle PDU Session Establishment Failures     **
+**                                                                        **
+**                                                                        **
+***************************************************************************/
+M5GSmCause amf_smf_validate_ie_params(
+    amf_ue_ngap_id_t ue_id, ULNASTransportMsg* msg) {
+  std::shared_ptr<smf_context_t> smf_ctx;
+  M5GSmCause cause = M5GSmCause::INVALID_CAUSE;
+  /*
+  Cause #27 – Missing or unknown DNN
+  the external DNN because the DNN was not included
+  although required or if the DNN could not be resolved.
+  */
+  if (msg->dnn.dnn.empty()) {
+    cause = M5GSmCause::MISSING_OR_UNKNOWN_DNN;
+  }
+
+  /*
+  Cause #28 – Unknown PDU session type
+  the requested PDU session type could not be recognized or is not allowed.
+  */
+  M5GPduSessionType session_type = static_cast<M5GPduSessionType>(
+      msg->payload_container.smf_msg.msg.pdu_session_estab_request
+          .pdu_session_type.type_val);
+  if (M5GPduSessionType::UNSTRUCTURED == session_type ||
+      M5GPduSessionType::ETHERNET == session_type) {
```

Comment:
Please use the variable on the left hand side of comparison for readability
```suggestion
  if (session_type == M5GPduSessionType::UNSTRUCTURED  ||
      session_type ==  M5GPduSessionType::ETHERNET) {
```

Reply:
Addressed review comment.